application-task-manager-api/src/main/java/com/xwiki/taskmanager/internal/TaskMacroUpdateEventListener.java

```diff
@@ -0,0 +1,187 @@
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.xwiki.bridge.event.DocumentCreatingEvent;
+import org.xwiki.bridge.event.DocumentUpdatingEvent;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.EntityType;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.EntityReferenceSerializer;
+import org.xwiki.model.reference.SpaceReference;
+import org.xwiki.query.Query;
+import org.xwiki.query.QueryException;
+import org.xwiki.query.QueryManager;
+import org.xwiki.rendering.block.XDOM;
+import org.xwiki.security.authorization.AuthorizationManager;
+import org.xwiki.security.authorization.Right;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+import com.xwiki.taskmanager.TaskCounter;
+import com.xwiki.taskmanager.model.Task;
+
+/**
+ * Listener that will create/modify the Task pages associated with the TaskMacros inside a page.
+ *
+ * @version $Id$
+ * @since 1.0
+ */
+@Component
+@Named("com.xwiki.taskmanager.internal.TaskMacroUpdateEventListener")
+@Singleton
+public class TaskMacroUpdateEventListener extends AbstractTaskEventListener
+{
+    private static final String TASK_PAGE_PREFIX = "Task";
+
+    @Inject
+    private Provider<TaskCounter> taskCounter;
+
+    @Inject
+    private QueryManager queryManager;
+
+    @Inject
+    private AuthorizationManager authorizationManager;
+
+    @Inject
+    @Named("compactwiki")
+    private EntityReferenceSerializer<String> serializer;
+
+
+    /**
+     * Default constructor.
+     */
+    public TaskMacroUpdateEventListener()
+    {
+        super(TaskMacroUpdateEventListener.class.getName(), Arrays.asList(new DocumentUpdatingEvent(),
+            new DocumentCreatingEvent()));
+    }
+
+    @Override
+    protected void processEvent(XWikiDocument document, XWikiContext context)
+    {
```

Comment:
We want the `TaskObjectUpdateEventListener` to trigger `TaskMacroUpdateEventListener` so it will update the `description` of the Task Object. 

The flow would look like this:
1. The user modifies and saves a Task Page. All the fields can be updated except the `description` because it is the macro call.
2. The Macro Page listener is triggered, sets the flag and uses the new values (especially the `assignee` and `deadline`) to render the macro and set the `description` field for the modified object.
3. The Task Page listener sees the flag and doesn't run again.